### PR TITLE
fix incorrect example for profile diff command.

### PR DESCRIPTION
**Pull request by morvencao**
fix the incorrect message for `istioctl profile diff` command for https://github.com/istio/istio/issues/29677

[ ] Configuration Infrastructure
[ ] Docs
[ X ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Review by carolynhu — APPROVED** [review]

File: operator/cmd/mesh/profile-diff.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ func profileDiffCmd(rootArgs *rootArgs, pfArgs *profileDiffArgs) *cobra.Command
 		Short: "Diffs two Istio configuration profiles",
 		Long:  "The diff subcommand displays the differences between two Istio configuration profiles.",
 		Example: `  # Profile diff by providing yaml files
-  istioctl profile diff manifests/profiles/default.yaml manifests/profiles/demo.yaml
+  istioctl profile diff --manifests manifests default demo
```
Comment (shamsher31) [inline operator/cmd/mesh/profile-diff.go:43]:
```suggestion
  istioctl profile diff default demo -d manifests
```

Comment (shamsher31) [reply operator/cmd/mesh/profile-diff.go:43]: This makes it clear that we are diffing two profile from manifests folder.

Comment (morvencao) [reply operator/cmd/mesh/profile-diff.go:43]: Both the commands work, right?

Comment (shamsher31) [reply operator/cmd/mesh/profile-diff.go:43]: Yes, both work fine. But I think we should use `command > subcommand > main flags > sub flags`.

Comment (shamsher31) [reply operator/cmd/mesh/profile-diff.go:43]: Does the description also need to be updated?
`# Profile diff by providing yaml files`

Comment (morvencao) [reply operator/cmd/mesh/profile-diff.go]: Thanks, addressed comments.

**Review by shamsher31 — APPROVED** [review]

File: operator/cmd/mesh/profile-diff.go [hunk 2]
```diff
@@ -39,11 +39,11 @@ func profileDiffCmd(rootArgs *rootArgs, pfArgs *profileDiffArgs) *cobra.Command
...
 		Long:  "The diff subcommand displays the differences between two Istio configuration profiles.",
-		Example: `  # Profile diff by providing yaml files
-  istioctl profile diff manifests/profiles/default.yaml manifests/profiles/demo.yaml
+		Example: `  # Profile diff by providing profile names
+  istioctl profile diff default demo
 
-  # Profile diff by providing a profile name
-  istioctl profile diff default demo`,
+  # Profile diff by providing manifests path and profile names
+  istioctl profile diff default demo -d manifests/`,
```
Comment (richardwxn) [inline operator/cmd/mesh/profile-diff.go:46]: this is needed only for dev builds

Comment (morvencao) [reply operator/cmd/mesh/profile-diff.go:46]: not only for dev build, especially useful if customized chart is specified.

**Pull request closed without merging**